server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateAttributesManager.java

```diff
@@ -121,24 +121,41 @@ public CheckedRunnable<IOException> getAsyncMetadataReadAction(
         LatchedActionListener<RemoteReadResult> listener
     ) {
         final ActionListener actionListener = ActionListener.wrap(
-            response -> listener.onResponse(new RemoteReadResult((ToXContent) response, CLUSTER_STATE_ATTRIBUTE, component)),
+            response -> listener.onResponse(new RemoteReadResult(response, CLUSTER_STATE_ATTRIBUTE, component)),
             listener::onFailure
         );
         return () -> getStore(blobEntity).readAsync(blobEntity, actionListener);
     }
 
-    public Map<String, ClusterState.Custom> getUpdatedCustoms(ClusterState clusterState, ClusterState previousClusterState) {
-        Map<String, ClusterState.Custom> updatedCustoms = new HashMap<>();
-        Set<String> currentCustoms = new HashSet<>(clusterState.customs().keySet());
-        for (Map.Entry<String, ClusterState.Custom> entry : previousClusterState.customs().entrySet()) {
-            if (currentCustoms.contains(entry.getKey()) && !entry.getValue().equals(clusterState.customs().get(entry.getKey()))) {
-                updatedCustoms.put(entry.getKey(), clusterState.customs().get(entry.getKey()));
-            }
-            currentCustoms.remove(entry.getKey());
+    public DiffableUtils.MapDiff<String, ClusterState.Custom, Map<String, ClusterState.Custom>> getUpdatedCustoms(
+        ClusterState clusterState,
+        ClusterState previousClusterState,
+        boolean includeEphemeral,
+        boolean firstUploadForEphemeralMetadata
```

Comment:
isWriteFull - ClusterState.Custom is all ephemeral right?

Reply:
changed

Reply:
Updated